pyethapp/tests/test_rpc_client.py

```diff
@@ -18,3 +83,68 @@ def test_set_host():
     client = JSONRPCClient(host)
     assert client.transport.endpoint == 'http://{}:{}'.format(host, client.port)
     assert client.transport.endpoint != 'http://{}:{}'.format(default_host, client.port)
+
+# The fixture takes much time to initialize, so the tests are grouped into one method
+def test_client(test_setup):
+    (test_app, rpc_tests_dir) = test_setup
+    client = JSONRPCClient(port=8081)
+
+    genesis_block_info = client.call('eth_getBlockByNumber', 'earliest', False)
+    genesis_gas_limit = quantity_decoder(genesis_block_info['gasLimit'])
+    assert client.default_tx_gas == (genesis_gas_limit - 1)
+
+    sender = client.sender
+    assert sender == '\xde\x0b)Vi\xa9\xfd\x93\xd5\xf2\x8d\x9e\xc8^@\xf4\xcbi{\xae'
+
+    coinbase = client.coinbase
+    assert coinbase == '\xde\x0b)Vi\xa9\xfd\x93\xd5\xf2\x8d\x9e\xc8^@\xf4\xcbi{\xae'
+
+    blocknumber = client.blocknumber()
+    assert blocknumber == 32
+
+    nonce = client.nonce(sender)
+    assert nonce == 0
+
+    balance1 = client.balance(sender)
+    assert balance1 == 5156250000000000000
+
+    gaslimit = client.gaslimit()
+    assert gaslimit == 3141592
+
+    lastgasprice = client.lastgasprice()
+    assert lastgasprice == 1
+
+    balance2 = client.balance('\xff' * 20)
+    assert balance2 == 0
+    fid = client.new_filter('pending', 'pending')
+
+    # The following tests require an account with a positive balance
+    # accs = client.call('eth_accounts')
+    # sender = accs[0]
+    # res_est = client.eth_sendTransaction(nonce, sender, address_encoder('\xff' * 20), 1)
+    # assert 'result' in res_est.keys()
+
+    # res_call = client.eth_call(utils.encode_hex(a0), '\xff' * 20, 0)
+    # assert 'result' in res_call.keys()
+
+    # res_st = client.send_transaction(sender, address_encoder('\xff' * 20), 1)
+    # assert 'result' in res_st.keys()
+
+    # solidity_code = "contract test { function multiply(uint a) returns(uint d) {   return a * 7;   } }"
+
+    # import ethereum._solidity
+    # s = ethereum._solidity.get_solidity()
+    # if s is None:
+    #     pytest.xfail("solidity not installed, not tested")
+    # else:
+    #     abi = s.mk_full_signature(solidity_code)
+    #     abic = client.new_abi_contract(abi, sender)
+    #     mult = abic.multiply(11111111)
+    #     assert mult == 77777777
+
```

Comment:
Yes, it can be updated. A while ago you sent me the authorization hash for validating transactions with non-zero transfers, but it remained on slack unlike me. Could you please remind what's been there? You can send me the whole chat log so you don't have to search for it, thank you